experimental/tiledb/common/dag/data_block/data_block.h

```diff
@@ -279,21 +289,22 @@ class DataBlockImpl {
   }
 };
 
-template <class Allocator>
-Allocator DataBlockImpl<Allocator>::allocator_;
-
 /**
  * The `DataBlock` class is the `DataBlockImpl` above, specialized to the
  * PoolAllocator with chunk size specified by our defined `chunk_size_`.
  */
-using DataBlock = DataBlockImpl<PoolAllocator<chunk_size_>>;
+template <size_t block_size = 64 * 1'024>
+class DataBlock : public DataBlockImpl<block_size> {
+  using Base = DataBlockImpl<block_size>;
+  using Base::Base;
+};
 
 /**
  * Function for getting new `DataBlock`s
  */
-[[maybe_unused]] static DataBlock make_data_block(size_t init_size) {
-  return DataBlock{init_size};
-}
+// [[maybe_unused]] static DataBlock make_data_block() {
```

Comment:
remove if it won't be reintroduced in the future or just `#if 0` plus comment for consistency.